### ISSUE TITLE
[SEVERE] hive_generator:hive_generator on lib/view/provider.dart (cached):  You have to provide a non-null typeId.

### ISSUE THREAD

**Issue by berdibekovaG** [hivedb/hive#637]
when I do "flutter pub run build_runner build --delete-conflicting-outputs"There were error  [SEVERE] hive_generator:hive_generator on lib/view/provider.dart (cached):
You have to provide a non-null typeId. how can I fix it? 

import 'package:hive/hive.dart';
part 'history_item.g.dart';

@HiveType(typeId: 0)
class HistoryItem{
  @HiveField(0)
  final String expression;
  @HiveField(1)
  final String result;

  const HistoryItem(this.expression, this. result);
}

**Version**
version: 1.0.0+1

environment:
  sdk: ">=2.7.0 <3.0.0"

dependencies:
  flutter:

    sdk: flutter
  cupertino_icons: ^1.0.2
  math_expressions: ^2.1.0
  shared_preferences: ^2.0.5
  provider: ^4.1.3
  hive: ^2.0.3
  hive_flutter: ^1.0.0

dev_dependencies:
  hive_generator: ^1.0.1
  build_runner: ^1.12.2

**Comment by sc-ot** [comment]
Did you solve it? My model looks kinda same, but i get the error: Null check operator used on a null value,. My Model:

@HiveType(typeId: 1)
class RequestData {
  @HiveField(0)
  Network network;
  @HiveField(1)
  Response<dynamic> response;
  @HiveField(2)
  String path;
  @HiveField(3)
  Method method;
  @HiveField(4)
  Function(dynamic) parser;
  @HiveField(5)
  bool cacheRequest;
  @HiveField(6)
  dynamic body;
  @HiveField(7)
  dynamic headers;
  @HiveField(8)
  dynamic queryParams;
  @HiveField(9)
  StreamController resultController;

  RequestData(
      this.network,
      this.response,
      this.path,
      this.method,
      this.parser,
      this.cacheRequest,
      this.resultController,
      this.body,
      this.headers,
      this.queryParams);
}

**Comment by berdibekovaG** [comment]
I don't resolve it but actually it's work's ok with this error. Just run
our code

сб, 17 апр. 2021 г., 22:33 sc-ot ***@***.***>:

**Comment by TheMisir** [comment]
The original issue needs a bit investigation to find cause of the error.

But @sc-ot doing lots of things that doesn't had to be done. You can't store runtime types like methods (`  @HiveField(4)
  Function(dynamic) parser;`) or complex types like StreamController on Hive or any sort of database. Databases meant to be used for storing data & value types.

**Comment by sc-ot** [comment]
Yes, was my bad. Thought it was possible with Hive. I already changed it.

**Comment by jbjason** [comment]
i had the same issue , u just need to do following steps--
1. run the command multiple times
2. change the typeId to @HiveType(typeId: 1) or @HiveType(typeId: 01)
3. run the command without any typeId like @HiveType()...
4. restart ur IDE.. 
hope one of them will definitely solve the issue :)
